celer/tests/test_homotopy.py

```diff
@@ -142,3 +160,26 @@ def test_zero_column(sparse_X):
     np.testing.assert_array_less(gaps[-1], tol)
     np.testing.assert_equal(w.shape[0], X.shape[1])
     np.testing.assert_equal(theta.shape[0], X.shape[0])
+
+
+@pytest.mark.parametrize("sparse_X", [False, True])
+def test_warm_start(sparse_X):
+    """Test Lasso path convergence."""
+    X, y, _, _ = build_dataset(
+        n_samples=100, n_features=1000, sparse_X=sparse_X)
```

Comment:
@Malcolm this is too big, the test will run slowly. Use smaller p0 if you want to guarantee multiple iterations